static/src/javascripts-legacy/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -124,37 +115,36 @@ define([
          *
          */
         function deriveBannerParams() {
-            var paramsByOfferingForUserEdition = editionParams[config.page.edition];
-
-            var engagementBannerTest = find(MembershipEngagementBannerTests, function(test) {
-                return ab.testCanBeRun(test) && segmentUtil.isInTest(test)
-            });
-
-            var userVariant = engagementBannerTest ? segmentUtil.variantFor(engagementBannerTest) : undefined;
+            var defaultParams = membershipEngagementBannerUtils.defaultParams;
+            var userTest = getUserTest();
+            var userVariant = getUserVariant(userTest);
+            var userVariantParams = {};
```

Comment:
Rather than mutating `userVariantParams`, could you rewrite the logic below to be 2 separate functions ? ie 
```
function getUserVariantParams(userVariant) {
  if (userVariant && userVariant.engagementBannerParams) {
    var params = userVariant.engagementBannerParams;
   ...
    return params;
  } else {
    return {};
  }
}

if (userVariant && userVariant.blockEngagementBanner) { 
    return DO_NOT_RENDER_ENGAGEMENT_BANNER;
}

return assign({}, defaultParams, getUserVariantParams(userVariant));